src/rviz/message_filter_display.h

```diff
@@ -58,6 +58,10 @@ Q_OBJECT
       topic_property_ = new RosTopicProperty( "Topic", "",
                                               "", "",
                                               this, SLOT( updateTopic() ));
+      unreliable_property_ = new BoolProperty( "Unreliable", false,
```

Comment:
I believe this code handles the deleting: https://github.com/ros-visualization/rviz/blob/indigo-devel/src/rviz/properties/property.cpp#L84-L98

It's definitely a round about way to do it.